internal/controller/authenticator_test.go

```diff
@@ -312,3 +318,92 @@ func (suite *ControllerTestSuite) TestParseToken() {
 	_, err = ParseToken("invalid-token")
 	assert.Error(suite.T(), err)
 }
+
+func (suite *ControllerTestSuite) TestHTTPProxySupport() {
+	// Start a mock proxy server
+	proxyServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write([]byte("Proxy used")); err != nil {
+			suite.T().Errorf("Failed to write response: %v", err)
+		}
+	}))
+	defer proxyServer.Close()
+
+	// Set the HTTPS_PROXY environment variable to the mock proxy
+	os.Setenv("HTTPS_PROXY", proxyServer.URL)
```

Comment:
Never modify or rely on environment variables in unit tests, because this prevents parallelizing unit tests.